### PR TITLE
Split `grangercausalitytests()`` into function that runs on only one `lag` and one `type` of test.

### PR DESCRIPTION
**Pull request by lababidi**
The current grangercausalitytests() runs a for-loop over all lag values from 1 to maxlag. This is a new function that accepts a single lag value and only computes the granger causality for that lag-value. Additionally this function accepts a type of test that is to be run. This will allow a user to choose which lag values and types of tests that need to be run.

### CODE REVIEW

**Comment by bashtage** [comment]
There are a number of issues:

1. The code is repetitve.  Why not just add one parameter to the existing GC code that would instruct which behavior?  Such as 

```
def grangercausalitytests(x, maxlag, addconst=True, verbose=True, all_lags=True):
        all_lags : bool
            Flag indicating whether to test all lags between 1 and maxlag (True) or only 
            maxlag (False)
```

2.  The docstring is poorly formatted.  Not quite your fault since the original one is also poorly formatted, but this should be cleaned up here.

File: statsmodels/tsa/stattools.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 from .adfvalues import mackinnonp, mackinnoncrit
 from statsmodels.tsa.arima_model import ARMA
 from statsmodels.compat.scipy import _next_regular
+import logging
+
+logger = logging.getLogger('statsmodels')
```
Comment (bashtage) [inline statsmodels/tsa/stattools.py]: No logging in accepted code

File: statsmodels/tsa/stattools.py [hunk 2]
```diff
@@ -735,6 +738,135 @@ def levinson_durbin(s, nlags=10, isacov=False):
     pacf_[0] = 1.
     return sigma_v, arcoefs, pacf_, sig, phi  # return everything
 
+def granger_causality(x, lag, addconst=True):
+    """four tests for granger non causality of 2 timeseries
+
```
Comment (bashtage) [inline statsmodels/tsa/stattools.py]: Needs correct sentence capitalization throughout

File: statsmodels/tsa/stattools.py [hunk 3]
```diff
@@ -735,6 +738,135 @@ def levinson_durbin(s, nlags=10, isacov=False):
...
+        all test results, dictionary keys are the number of lags. For each
+        lag the values are a tuple, with the first element a dictionary with
+        teststatistic, pvalues, degrees of freedom, the second element are
+        the OLS estimation results for the restricted model, the unrestricted
+        model and the restriction (contrast) matrix for the parameter f_test.
+
+    Notes
+    -----
+    TODO: convert to class and attach results properly
+
```
Comment (bashtage) [inline statsmodels/tsa/stattools.py]: Move TODO to function comment

File: statsmodels/tsa/stattools.py [hunk 4]
```diff
@@ -735,6 +738,135 @@ def levinson_durbin(s, nlags=10, isacov=False):
...
+    result['params_ftest'] = (np.squeeze(ftres.fvalue)[()],
+                          np.squeeze(ftres.pvalue)[()],
+                          ftres.df_denom, ftres.df_num)
+
+    result['models'] = [res2down, res2djoint, rconstr]
+
+    return result
 
 def grangercausalitytests(x, maxlag, addconst=True, verbose=True):
     """four tests for granger non causality of 2 timeseries
```
Comment (bashtage) [inline statsmodels/tsa/stattools.py]: Make an additional parameter and just special case here

**Comment by josef-pkt** [comment]
I also think adding a second function just to select some option is too much code duplication. It should be possible to add this through options to the existing function.

However, I think it would be useful to add a second function that runs many granger causality tests and avoids recreating the lag matrices for each one separately. For example, if we want to run all pairwise granger causality tests for a large number of time series with fixed number of lags.

**Pull request closed without merging**